### DISCUSSION TITLE
[Improving EventListener Provider]

### DISCUSSION THREAD

**Discussion by taufanvps22** [keycloak/keycloak#24989] [Ideas]
Hi folks wanted to discuss related the weird behavior in my opinion about how we currently  handle event.

in [EventBuilder](https://github.com/keycloak/keycloak/blob/68b33be65536bb98af8c60e3207ee451040d5b3c/server-spi-private/src/main/java/org/keycloak/events/EventBuilder.java#L264) currently for `EventStoreProvider` we do  check enabled events from Realm configuration, then why we don't do the same  in `EventListenerProvider` as well ?

Currently we just found out discrepancy between EventStoreProvider. (JDBC) and EventListenerProvider (custom publishing via kafka), I know we can just do filtering on each specific EventListenerProvider but in term of design wise, will be good to have that filtering applied for EventListenerProvider as well, it will give us more control from admin UI and more proper.

wdyt ? 
I'll Create a PR if this idea sounds good

Thnak you

**Comment by danielFesenmeyer** [comment]
Currently there is no possibility to restrict the types of events for listeners. You can decide in your listener, whether the event is relevant or not. You could even use the same logic as used for the EventStoreProvider, if needed.

But it would not make sense to enforce the same restrictions for EventStoreProvider and EventListenerProvider: You might want to listen to an event (and for example send an email under a certain condition), but you might not want to have this event stored. (And it makes sense not to store every event in the Keycloak storage, because it's not an optimized event storage and it will become slow with too many events.)

**Reply by taufanvps22** [reply]
Thank u for the clarification @danielFesenmeyer
